examples/src/components/DrillWithExternalDataExample.jsx

```diff
@@ -0,0 +1,378 @@
+import {
+    projectId,
+    employeeNameIdentifier,
+    averageDailyTotalSales,
+    locationStateDisplayFormIdentifier,
+    locationStateAttributeUri,
+    totalSalesIdentifier,
+    locationNameDisplayFormIdentifier,
+    locationNameAttributeUri
+} from '../utils/fixtures';
+
+const EmployeeProfile = ({ gender, dob, cell, id, registered, location }) => {
+    const menOrWomen = gender === 'male' ? 'men' : 'women';
+    return (
+        <div className="employeeProfile s-employee-profile">
+            {/* language=CSS */}
+            <style jsx>{`
+                .employeeProfile {
+                    display: flex;
+                    margin-top: 20px;
+                }
+                .picture {
+                    border-radius: 50%;
+                    border: 2px solid #14b2e2;
+                    width: 100px;
+                    height: 100px;
+
+                }
+                .text {
+                    flex: 1 1 auto;
+                    margin-left: 20px;
+                }
+            `}</style>
+            <div><img className="picture" src={`https://randomuser.me/api/portraits/${menOrWomen}/${parseInt(id, 10) % 100}.jpg`} alt="" /></div>
+            <div className="text" >
+                <p>Date of birth: {dob}</p>
+                <p>Place of birth: {location.city}, {location.state}</p>
+                <p>Phone: {cell}</p>
+                <p>Employed since: {registered}</p>
+            </div>
+        </div>
+    );
+};
+EmployeeProfile.propTypes = {
+    gender: PropTypes.string.isRequired,
+    dob: PropTypes.string.isRequired,
+    cell: PropTypes.string.isRequired,
+    location: PropTypes.object.isRequired,
+    registered: PropTypes.string.isRequired,
+    id: PropTypes.string.isRequired
+};
+
+export class DrillWithExternalDataExample extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            state: null,
+            location: null,
+            employee: null,
+            employee3rdPartyData: {
+                error: null,
+                isLoading: false,
+                data: null
+            }
+        };
+    }
+
+    onLocationDrill = (drillTarget) => {
+        const { title: name, id } = drillTarget.drillContext.element === 'bar'
+            ? drillTarget.drillContext.intersection[1]
+            : drillTarget.drillContext.points[0].intersection[1];
+        const location = {
+            name,
+            uri: `${locationNameAttributeUri}/elements?id=${id}`
+        };
+        this.setState({
+            location,
+            state: null
+        });
+    }
+
+    onStateDrill = (drillTarget) => {
+        const { name, id } = drillTarget.drillContext.row[0];
+        const state = {
+            name,
+            uri: `${locationStateAttributeUri}/elements?id=${id}`
+        };
+        this.setState({
+            state,
+            location: null
+        });
+    }
+
+    onClearState = () => {
```

Comment:
I'd keep the naming consistent, so if there are onLocationDrill and onStateDrill, I'd use onStateClear and onLocationClear or vice versa